crates/sui-graphql-rpc/src/mutation.rs

```diff
@@ -71,27 +78,72 @@ impl Mutation {
             );
         }
         let transaction = Transaction::from_generic_sig_data(tx_data, sigs);
+        let options = SuiTransactionBlockResponseOptions::new()
+            .with_events()
+            .with_balance_changes()
+            .with_raw_input()
+            .with_raw_effects();
 
         let result = sui_sdk_client
             .quorum_driver_api()
             .execute_transaction_block(
                 transaction,
-                SuiTransactionBlockResponseOptions::default(),
-                Some(ExecuteTransactionRequestType::WaitForEffectsCert),
+                options,
+                // This needs to be WaitForLocalExecution because we need the transaction effects.
+                // TODO: make it possible to execute without waiting for local execution?
+                Some(ExecuteTransactionRequestType::WaitForLocalExecution),
             )
             .await
             // TODO: use proper error type as this could be a client error or internal error
             // depending on the specific error returned
             .map_err(|e| Error::Internal(format!("Unable to execute transaction: {e}")))
             .extend()?;
 
+        let raw_effects: NativeTransactionEffects = bcs::from_bytes(&result.raw_effects)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction effects: {e}")))
+            .extend()?;
+        let sender_signed_data: SenderSignedData = bcs::from_bytes(&result.raw_transaction)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))
+            .extend()?;
+
+        let events = result
+            .events
+            .ok_or(Error::Internal(
+                "No events are returned from tranasction execution".to_string(),
+            ))?
+            .data
+            .into_iter()
+            .map(|e| Event {
+                stored: None,
+                native: NativeEvent {
+                    package_id: e.package_id,
+                    transaction_module: e.transaction_module,
+                    sender: e.sender,
+                    type_: e.type_,
+                    contents: e.bcs,
+                },
+            })
+            .collect();
+
+        let balance_changes = result.balance_changes.ok_or(Error::Internal(
+            "No balance changes are returned from tranasction execution".to_string(),
```

Comment:
Nit transaction (altho this is exactly how I tend to misspell it too so I feel that)

Reply:
Oops and I pasted this to a few other places too..